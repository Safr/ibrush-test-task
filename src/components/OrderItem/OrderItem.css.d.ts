export const OrderItem: string;
export const orderItem: string;
export const Up: string;
export const up: string;
export const Down: string;
export const down: string;
