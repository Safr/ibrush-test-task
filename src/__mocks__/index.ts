export const orderBook = {
  BTC_USD: {
    ask: ['6891', '0.005', '34.455'],
    ask_amount: '2413543.16939829',
    ask_quantity: '190.96765367',
    ask_top: '6887.20157116',
    bid: ['6849.7394927', '0.75', '5137.30461952'],
    bid_amount: '286086.48310063',
    bid_quantity: '11524.76654836',
    bid_top: '6875',
  },
};

export const order = {
  avg: '6875.86448007',
  buy_price: '6859.227881',
  high: '6969.54201841',
  last_trade: '6864.01',
  low: '6800',
  sell_price: '6873.71642794',
  updated: 1528752429,
  vol: '1112.01492053',
  vol_curr: '7632881.53471028',
};

export const currentOrder = {
  BTC_USD: {
    avg: '6875.86448007',
    buy_price: '6859.227881',
    high: '6969.54201841',
    last_trade: '6864.01',
    low: '6800',
    sell_price: '6873.71642794',
    updated: 1528752429,
    vol: '1112.01492053',
    vol_curr: '7632881.53471028',
  },
};

export const orders = { currentOrder };
export const changedOrders = [''];
export const orderBookInfo = [['5897.8242', '0.003', '17.6934726']];

// export const fetchOrders = () => Promise.resolve({
//   orders: {
//     BTC_USD: {
//       avg: '6875.86448007',
//       buy_price: '6859.227881',
//       high: '6969.54201841',
//       last_trade: '6864.01',
//       low: '6800',
//       sell_price: '6873.71642794',
//       updated: 1528752429,
//       vol: '1112.01492053',
//       vol_curr: '7632881.53471028',
//     },
//   },
// });
