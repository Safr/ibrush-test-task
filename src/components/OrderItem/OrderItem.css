.OrderItem {
  width: 100px;
  margin: 10px;
  padding: 7px;
  background-color: #fff;
  border-radius: 255px 15px 225px 15px/15px 225px 15px 255px;
  border: solid 2px #41403E;
  cursor: pointer;
  transform: translateY(0);
  transition: all .3s ease;
}

.OrderItem:hover,
.OrderItem:active {
  transform: translateY(3px);
  border:solid 1px #41403E;
}

span.Up::before {
  content: '';
  display: inline-block;
  width: 0;
  height: 0;
  margin-right: 2px;
  border-style: solid;
  border-width: 0 7px 12px 7px;
  border-color: transparent transparent #57cc1d transparent;
}

span.Down {
  content: '';
  display: inline-block;
  width: 0;
  height: 0;
  margin-right: 2px;
  border-style: solid;
  border-width: 0 7px 12px 7px;
  border-color: transparent transparent #f56151 transparent;
}


p.Up {
  color: #57cc1d;
}

p.Down::before {
  color: #f56151;
}