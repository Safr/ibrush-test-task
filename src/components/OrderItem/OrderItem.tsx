import * as React from 'react';
import classes from './OrderItem.css';
import { OrderInterface } from '../../types';

interface Props {
  orderName: string;
  order: OrderInterface;
  changedPrice: string;
  onOpenPopup: (orderName: string, className: string, order: OrderInterface) => void;
}

const OrderItem: React.SFC<Props> = ({ orderName, order, changedPrice, onOpenPopup }) => {
  const handleOpenPopup = function () {
    onOpenPopup(orderName, className, order);
  };
  const className = changedPrice === 'up' ? 'Up'
    : changedPrice === 'down' ? 'Down' : '';
  return (
    <li className={classes.OrderItem}>
      <a onClick={handleOpenPopup}>
        <p className={classes[`${className}`]}>{orderName}</p>
        <span className={classes[`${className}`]}>{parseFloat(order.buy_price).toFixed(2).replace('.', ',')}</span>
      </a>
    </li>
  );
};

export default OrderItem;
