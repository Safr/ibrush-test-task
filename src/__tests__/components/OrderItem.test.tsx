import React from 'react';
import { shallow } from 'enzyme';
import OrderItem from '../../components/OrderItem/OrderItem';
import { order } from '../../__mocks__';

describe('OrderItem Component', () => {
  let wrapper;
  let onOpenPopup;
  beforeEach(() => {
    onOpenPopup = jest.fn();
    wrapper = shallow(
      <OrderItem
        orderName="BTC_USD"
        order={order}
        changedPrice=""
        onOpenPopup={onOpenPopup}
      />,
    );
  });
  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should handle onOpenPopup', () => {
    wrapper.find('a').simulate('click');
    expect(onOpenPopup).toHaveBeenCalledWith('BTC_USD', '', order);
  });
});
