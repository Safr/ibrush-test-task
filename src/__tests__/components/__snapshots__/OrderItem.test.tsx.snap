// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderItem Component should render correctly 1`] = `
<li>
  <a
    onClick={[Function]}
  >
    <p>
      BTC_USD
    </p>
    <span>
      6859,23
    </span>
  </a>
</li>
`;
